### PR TITLE
Des 52 scheduling bug on addition

### PR DESCRIPTION
**Pull request by AndrewLin95**
Originally, bugs occurred when scheduling and deleting items from the schedule.

**Addition**
- It was allowed to schedule 3 items on top of each other.

**Deletion**
- Deleting items produced unexpected side effects.

**_Changes_**
**Addition**
- Changed the conflict amount from `> 3` to `> 2`

**Deletion**
- Reworked logic to use a more general approach to delete reducing the number of scenarios to 2.
- In summary:
    1) All data that conflicts with the target data to delete was identified.
    2) For each of these conflicts, we recursively identify any subsequent conflicting data. 
        2a) If there are **no** conflicts, reset position and columns to default.
        2b) If there are conflicts, re-sort and re-assign position and columns.
    3) Delete old data, then add the new data.

### CODE REVIEW

**Review by skim2264 — APPROVED** [review]
I think it looks good! :D The code looks a lot cleaner with the changes too
